pkg/daemon/bootc.go

```diff
@@ -0,0 +1,222 @@
+type ImageReference struct {
+	Image     string          `json:"image"`
+	Transport string          `json:"transport"`
+	Signature *ImageSignature `json:"signature,omitempty"`
+}
+
+// ImageSignature mirrors https://github.com/containers/bootc/blob/94521ecb19b8d0f2bfc36220322c60cafa036296/lib/src/spec.rs#L57
+// Same as https://docs.rs/ostree-ext/latest/ostree_ext/container/enum.SignatureSource.html
+type ImageSignature struct {
+	AllowInsecure bool
+	OstreeRemote  string
+}
+
+// bootc.go
+
+// BootcClient provides all Bootc related methods in one structure.
+type BootcClient struct {
+	client Client
+}
+
+// Synchronously invoke bootc, writing its stdout to our stdout,
+// and gathering stderr into a buffer which will be returned in err
+// in case of error.
+func runBootc(args ...string) error {
+	return runCmdSync("bootc", args...)
+}
+
+func (b *BootcClient) Initialize() error {
+	// Commands like update and rebase need the pull secrets to pull images and manifests,
+	// make sure we get access to them when we Initialize
+	err := useMergedPullSecrets(bootcSystem)
+	if err != nil {
+		klog.Errorf("error while linking bootc pull secrets %v", err)
+	}
+	return nil
+}
+
+// GetBootedAndStagedImage returns the current booted and staged image found
+func (b *BootcClient) GetBootedAndStagedImage() (*BootEntry, *BootEntry, error) {
+	status, err := b.client.QueryStatus()
+	if err != nil {
+		return nil, nil, err
+	}
+	return status.Status.GetBootedImage(), status.Status.GetStagedImage(), nil
+}
+
+// // GetStatus returns multi-line human-readable text describing system status
+// // Bootc is working on status update: https://github.com/containers/bootc/issues/408
+// func (r *BootcClient) GetStatus() (string, error) {
+// 	output, err := runGetOut("bootc", "status")
+// 	if err != nil {
+// 		return "", err
+// 	}
+
+// 	return string(output), nil
+// }
+
+// GetBootcBootedOSImageURL returns the image URL as well as the image version(for logging) and the ostree commit (for comparisons)
+func (b *BootcClient) GetBootcBootedOSImageURL() (string, string, string, error) {
+	bootedImage, _, err := b.GetBootedAndStagedImage()
+	if err != nil {
+		return "", "", "", err
+	}
+	osImageURL := ""
+	if bootedImage.GetContainerImageReference() != "" {
+		osImageURL = bootedImage.GetContainerImageReference()
+	}
+	var baseChecksum string
+	if bootedImage.Ostree != nil {
+		baseChecksum = bootedImage.GetBaseChecksum()
+	} else {
+		baseChecksum = ""
+	}
+	return osImageURL, *bootedImage.Image.Version, baseChecksum, nil
+}
+
+// Switch target a new container image reference to boot for the system or errors if already swictehd.
+func (b *BootcClient) Switch(imgURL string) error {
+	// Try to re-link the merged pull secrets if they exist, since it could have been populated without a daemon reboot
+	if err := useMergedPullSecrets(bootcSystem); err != nil {
+		klog.Errorf("error while linking bootc pull secrets %v", err)
```

Comment:
question: same question here as above, do we just want to log the error and continue on?